com.avaloq.tools.ddk.xtext.ui.test/src/com/avaloq/tools/ddk/xtext/ui/templates/ResourceNameTemplateVariableResolverTest.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.ui.templates;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.jface.text.templates.TemplateException;
+import org.eclipse.jface.text.templates.TemplateVariable;
+import org.eclipse.xtext.XtextRuntimeModule;
+import org.eclipse.xtext.ui.editor.model.IXtextDocument;
+import org.eclipse.xtext.ui.editor.templates.XtextTemplateContext;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.google.common.collect.Iterables;
+import com.google.inject.Guice;
+
+
+public class ResourceNameTemplateVariableResolverTest {
+
+  private static final Object[] FILE = new Object[] {"file"}; //$NON-NLS-1$
+  private static final String FILENAME = "filename"; //$NON-NLS-1$
+
+  private static XtextTemplateContext mockContext;
+  private static IXtextDocument mockDocument;
+  private static IFile mockFile;
+
+  private static TemplateVariableResolverTestHelper helper;
+
+  private static ResourceNameTemplateVariableResolver resolver;
+
+  @BeforeClass
+  public static void beforeClass() {
+    mockContext = Mockito.mock(XtextTemplateContext.class);
+    mockDocument = Mockito.mock(IXtextDocument.class);
+    mockFile = Mockito.mock(IFile.class);
+
+    helper = Guice.createInjector(new XtextRuntimeModule()).getInstance(TemplateVariableResolverTestHelper.class);
+
+    resolver = new ResourceNameTemplateVariableResolver();
+
+    Mockito.when(mockContext.getDocument()).thenReturn(mockDocument);
+    Mockito.when(mockDocument.getAdapter(IFile.class)).thenReturn(mockFile);
+  }
+
+  @AfterClass
+  public static void afterClass() {

```

Comment:
hmm, this method looks quite weird. Is it possible to use our BeforeAll annotation for initialization method? This probably will require using com.avaloq.tools.ddk.test.core.junit.runners.ClassRunner

Reply:
Done.